gen_allocdefs.h

```diff
@@ -14,7 +14,11 @@ static int ta_rplus ## _internal (ta *x, unsigned int n, unsigned int pluslen) \
       return 1; } \
     return 1; } \
   x->len = 0; \
-  return !!(x->field = (type *) alloc((x->a = n) * sizeof(type))); } \
+  x->field = alloc(n * sizeof(type)); \
```

Comment:
How do we know the `(type *)` cast is no longer needed?

Reply:
Yes, this only works properly when alloc is defined as `void *alloc(...)`, but it is `char *alloc(...)`.
alloc.c should be changed to use the malloc prototype.

Reply:
You are right. I was under the assumption that ```alloc()``` behaves like ```malloc()``` and returns void*. It does not.